src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/ProxyWhitelistTest.java

```diff
@@ -66,4 +73,31 @@ public void caching() throws Exception {
         assertEquals(2, ((EnumeratingWhitelist)pw2.delegates.get(0)).permittedCache.size());
     }
 
+    /**
+     * Test concurrent modification of delegates when initializing a ProxyWhitelist. This may cause concurrent threads 
+     * to enter an infinite loop when using a {@link java.util.WeakHashMap} to hold the delegates as it is not 
+     * synchronized. This is a rare case that is also related to Garbage Collection.
+     * If contention is acceptable and/or the delegates collection behave, this test should be quite fast, just a few 
+     * seconds. A timeout is allowed to process the creation of 1000000 {@link ProxyWhitelist} using an original 
+     * delegate. If the tasks have not completed by then, we can assume that there is a problem.
+     */
+    @Issue("JENKINS-41797")
+    @Test(timeout = 30000)
+    public void testConcurrent() throws InterruptedException, IOException {
```

Comment:
Did this test reproduce the issue before you made the product change?

Reply:
It did yes. In the first build: https://ci.jenkins.io/blue/organizations/jenkins/Plugins%2Fscript-security-plugin/detail/PR-356/1/tests/.
Then the fix kicked off build 2 and everything passes